numpy/dow_selection/dow_selection.py

```diff
@@ -0,0 +1,75 @@
+"""
+Dow Selection
+-------------
+
+Topics: Boolean array operators, sum function, where function, plotting.
+
+The array 'dow' is a 2-D array with each row holding the
+daily performance of the Dow Jones Industrial Average from the
+beginning of 2008 (dates have been removed for exercise simplicity).
+The array has the following structure::
+
+       OPEN      HIGH      LOW       CLOSE     VOLUME      ADJ_CLOSE
+       13261.82  13338.23  12969.42  13043.96  3452650000  13043.96
+       13044.12  13197.43  12968.44  13056.72  3429500000  13056.72
+       13046.56  13049.65  12740.51  12800.18  4166000000  12800.18
+       12801.15  12984.95  12640.44  12827.49  4221260000  12827.49
+       12820.9   12998.11  12511.03  12589.07  4705390000  12589.07
+       12590.21  12814.97  12431.53  12735.31  5351030000  12735.31
+
+0. The data has been loaded from a .csv file for you.
+1. Create a "mask" array that indicates which rows have a volume
+   greater than 5.5 billion.
+2. How many are there?  (hint: use sum).
+3. Find the index of every row (or day) where the volume is greater
+   than 5.5 billion. hint: look at the where() command.
+
+Bonus
+~~~~~
+
+1. Plot the adjusted close for *every* day in 2008.
+2. Now over-plot this plot with a 'red dot' marker for every
+   day where the volume was greater than 5.5 billion.
+
+See :ref:`dow-selection-solution`.
+"""
+
+from numpy import loadtxt, sum, where
+import matplotlib.pyplot as plt
+import os
+# Constants that indicate what data is held in each column of
+# the 'dow' array.
+OPEN = 0
+HIGH = 1
+LOW = 2
+CLOSE = 3
+VOLUME = 4
+ADJ_CLOSE = 5
+
+# 0. The data has been loaded from a .csv file for you.
+
+# 'dow' is our NumPy array that we will manipulate.
+file=os.path.dirname(__file__)+'/dow.csv'
+dow = loadtxt(file, delimiter=',')
+
+# 1. Create a "mask" array that indicates which rows have a volume
+#    greater than 5.5 billion.
+masklargeVolumes=dow[:,VOLUME]>5.5e9
```

Comment:
Remember from `refactor_inventory` that two of the things you want to avoid in your code are so-called "magic numbers" and duplication.  Here `5.5e9` is a "magic number", and it appears twice: on this line and line 61.  You can solve both problems by using a "constant":
```
HIGH_VOL = 5.5e9
masklargeVolumes=dow[:,VOLUME]>HIGH_VOL
```
There are not really constants in Python, since any name can be reapplied to anything else, but there _is_ a gentleman's agreement that names in ALL_CAPS are _meant_ to be constant.  That's why `VOLUME` and `ADJ_CLOSE` appear that way.